### ISSUE TITLE
Implement Maximum Fragment Length TLS extension

### ISSUE THREAD

**Issue by frozeus** [libressl-portable/portable#214]
It seems that Libressl doesn't support maximum fragment length negotiation extension : https://tools.ietf.org/html/rfc6066#page-8

A patch has been posted on openssl group https://groups.google.com/d/msg/mailing.openssl.dev/fQxXvCg1uQY/_g09FmmhvgwJ

Could you review and integrate it for 2.5.X release ? This will be very usefull for embedded clients who can't allocate big buffer in RAM.

Thanks,

**Comment by 4a6f656c** [comment]
Do you have an actual use case for this? If so, could you please provide further details.

**Comment by frozeus** [comment]
Yes instead of several call on a server i would like to receive one "big" json (around 8Ko) and parse it on the go. I am not able to allocate more than 2ko of RAM for TLS.

**Comment by bob-beck** [comment]
In what application?

**Comment by frozeus** [comment]
IOT device

**Comment by bob-beck** [comment]
So this is not a public application. Do you know if OpenSSL is taking that patch?

**Comment by frozeus** [comment]
The patch is planned to be merged post 1.10 : https://github.com/openssl/openssl/pull/1008

**Comment by rojer** [comment]
+1 on this. it is very important for IoT connectivity where 16K contiguous allocation requirement is a problem. OpenSSL already supports this.
